django_tex/filters.py

```diff
@@ -1,20 +1,46 @@
 from django.utils.formats import localize_input
 from django.template.defaultfilters import register
 
+BACKSLASH_REPLACEMENT = "<:-!~#BACKSLASH#~!:->"
 
-def do_linebreaks(value):
+
+def do_linebreaks(value: str) -> str:
     return value.replace("\n", "\\\\\n")
 
 
-def do_latex_escape(value):
+def do_latex_escape(value: str) -> str:
+    """
+    Replace all LaTeX characters that could cause the latex compiler to fail
+    and at the same time try to display the character as intended from the user.
+
+    see also https://tex.stackexchange.com/questions/34580/escape-character-in-latex
+    """
+
     return (
-        value.replace("&", "\\&")
-        .replace("$", "\\$")
-        .replace("%", "\\%")
-        .replace("#", "\\#")
-        .replace("_", "\\_")
-        .replace("{", "\\{")
-        .replace("}", "\\}")
+        # Replace Backslashes so we dont' double replace backslashes
+        value.replace("\\", BACKSLASH_REPLACEMENT)
+        # If there was a space before the character we assume the user wants
+        # it also in the displayed LaTeX text
+        .replace(BACKSLASH_REPLACEMENT + " ", r"\textbackslash\space ")
```

Comment:
Instead of pattern matching for the whitespace and inserting a `\space` command, we should probably use `{}` at the end of the replacement, e.g. `\` gets escaped to `\textbackslash{}`.

Reply:
agreed